infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/send/LogReplicationEventMetadata.java

```diff
@@ -74,6 +74,19 @@ public LogReplicationEventMetadata(UUID syncId, long syncTimestamp, long baseSna
         this.lastTransferredBaseSnapshot = baseSnapshot;
     }
 
+    /**
+     * Constructor
+     *
+     * @param syncId identifier of the request that preceded this event.
+     * @param syncTimestamp last synced timestamp.
+     * @param baseSnapshot last base snapshot
+     */
+    public LogReplicationEventMetadata(UUID syncId, long syncTimestamp, long baseSnapshot, boolean forcedSnapshotSync) {
```

Comment:
can we change the existing constructors to take the 'forced' flag instead of creating a new one?